Add visualization to show canisters per subnet

tarek-eg opened the pull request:
closes https://github.com/ic-rocks/ic-rocks/issues/18

Added cans per subnet donut chart.
The donut chart is not generic enough to be used for other use cases but this can be done once we have the need.
DataOverTimeChart is currently responsible for rendering other types of charts and needs refactoring in a seperate ticket maybe?

![image](https://user-images.githubusercontent.com/30959956/138225815-9ed77a77-890e-44bd-920e-19e07242505c.png)
![image](https://user-images.githubusercontent.com/30959956/138223831-33a23288-b0f8-4dc3-bb1f-a3ede9404786.png)

tarek-eg committed:
Add rechart package

tarek-eg committed:
Add donut chart component

tarek-eg committed:
Add canisters-per-subnet configs

tarek-eg committed:
Render cans per subnet chart

tarek-eg commented:
I was wondering where is the best place to put this chart?
For now I will put it with other `DataOverTime` chart so it appears in the chart page and when you click on the chart it takes you to the full view

tarek-eg commented:
The deployment is failing but I don't have the permissions to see why @FloorLamp can you check?

FloorLamp commented:
```
02:32:11.609 | Error occurred prerendering page "/charts". Read more: https://nextjs.org/docs/messages/prerender-error
-- | --
02:32:11.610 | Error: Minified React error #130; visit https://reactjs.org/docs/error-decoder.html?invariant=130&args[]=undefined&args[]= for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
02:32:11.610 | at a.b.render (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:45:32)
02:32:11.611 | at a.b.read (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:41:83)
02:32:11.611 | at exports.renderToString (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:52:138)
02:32:11.611 | at Object.renderPage (/vercel/path0/node_modules/next/dist/next-server/server/render.js:54:854)
02:32:11.612 | at Function.getInitialProps (/vercel/path0/.next/server/pages/_document.js:715:19)
02:32:11.612 | at loadGetInitialProps (/vercel/path0/node_modules/next/dist/next-server/lib/utils.js:5:101)
02:32:11.613 | at renderToHTML (/vercel/path0/node_modules/next/dist/next-server/server/render.js:54:1145)
02:32:11.613 | at async /vercel/path0/node_modules/next/dist/export/worker.js:26:6
02:32:11.614 | at async Span.traceAsyncFn (/vercel/path0/node_modules/next/dist/telemetry/trace/trace.js:6:584)
```

tarek-eg commented:
@FloorLamp ready for review

FloorLamp commented:
This is great! My only feedback is that the text color of `n canister` needs to be lighter in dark mode. Also `canister` should be pluralized (there's a `pluralize` helper function somewhere)

tarek-eg committed:
fix review comments
- add pluraization
- fix text color
- add link navigation to subnet throw chart

tarek-eg commented:
@FloorLamp 
I added the fix

FloorLamp reviewed (approved):
Awesome job!
